docs/components/INTRODUCTION.md

```diff
@@ -0,0 +1,77 @@
+# Component introduction
+
+A component refers to a modular, reusable piece of code that encapsulates specific functionality, styling, and behavior within a user interface (UI). Components are commonly used to build dynamic and interactive web applications, allowing developers to break down complex user interfaces into smaller, manageable parts.
+
+## How to generate a component
+
+```shell
+ng g component <ComponentName> [--project-name <project-name>]
+```
+
+## Structure
+Below you can see the basic files generated for a component (more files can be generated depending on the activated options):
+- `component-name.component.ts` will contain the logic of the component
+- `component-name.spec.ts` will contain component unit tests
+- `component-name.style.scss` will contain the styling of the component
+- `component-name.template.html` will contain the structure of the component
+
+> [!NOTE]
+> By convention, file names must be written in kebab-case
+
+
+## Component class
+Otter components are Angular components with Otter-specific information provided through the `@O3rComponent` decorator.
+```typescript
+@O3rComponent({
+  componentType: 'Component'
+})
+@Component({
+  selector: 'o3r-component-name',
+  standalone: true,
+  templateUrl: './component-name.template.html',
+  styleUrls: ['./component-name.style.scss'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ComponentNameComponent {}
+```
+
+> [!NOTE]
+> By convention, class names must be written in PascalCase and variable names in camelCase.
+
+### Component type
+The object passed to the `@O3rComponent` decorator includes the component type, which can be:
+- `Page`: a component that displays an application route
+- `Block`: a component that handles a functional area
+- `ExposedComponent`: a component that needs to be exposed in your CMS
+- `Component`: a component that does not need to be exposed in your CMS and that does not fit the others categories
+
+#### Example
+
+If you have an e-commerce application composed of different pages, like a shopping one.
+This `Page` will contain one or several blocks, like the one that displays the list of products.
+The `Block` is composed of different exposed components, such as the filtering component or the product card component.
+An `ExposedComponent` could be composed of several components such as the price `Component` inside the product card component.
```

Comment:
Not sure about this sentence, the exposed component will have it's configuration extracted and is considered as configurable, so for the expose component you can just give the example of a configuration from the filtering component that makes sense to be exposed for example. Then take an example such as the price component for the 'Component'